src/components/Search/SearchFiltersParticipantsSelector.tsx

```diff
@@ -104,17 +104,20 @@ function SearchFiltersParticipantsSelector({initialAccountIDs, onFiltersUpdate}:
             true,
         );
 
-        const isCurrentUserSelected = selectedOptions.find((option) => option.accountID === chatOptions.currentUserOption?.accountID);
-
         newSections.push(formattedResults.section);
 
-        if (chatOptions.currentUserOption && !isCurrentUserSelected) {
+        if (chatOptions.currentUserOption) {
+            const currentUserSelected = formattedResults.section.data.find((option) => option.accountID === chatOptions.currentUserOption?.accountID);
             const formattedName = ReportUtils.getDisplayNameForParticipant(chatOptions.currentUserOption.accountID, false, true, true, personalDetails);
-            newSections.push({
-                title: '',
-                data: [{...chatOptions.currentUserOption, text: formattedName}],
-                shouldShow: true,
-            });
+            if (!currentUserSelected) {
+                newSections.push({
+                    title: '',
+                    data: [{...chatOptions.currentUserOption, text: formattedName}],
+                    shouldShow: true,
+                });
+            } else {
+                currentUserSelected.text = formattedName;
```

Comment:
@FitseTLT 

We shouldn't directly mutate the `currentUserSelected` object alone which is the result of find operation and expect it to be updated in the `newSections` array.
Instead we should update the `formattedResults.section` before pushing to newSections.

Reply:
Yes it does get updated @abdulrahuman5196 because what `find` will return in this case is object (reference) and changes to it is going to be applied to newSections, in fact, that's why it works.; you can check it.

Reply:
@FitseTLT Yes. I was aware it was working and its using reference to work. But that's what I am suggesting as not a good code standard. Instead we should update the formattedResults.section before pushing to newSections.

Reply:
@abdulrahuman5196 the reason the pushing code is before the `text` updating code is to avoid `(you)` user becoming always on top as discussed [here](https://github.com/Expensify/App/issues/47712#issuecomment-2318500294)

Reply:
@FitseTLT I understand the issue and instead of going on this path we should try to refractor the code to fix the always on top issue. I think it could be simple refractoring. Could you kindly check on this.